src/Microsoft.TemplateEngine.Cli/Commands/BaseCommand.cs

```diff
@@ -0,0 +1,196 @@
+            IEngineEnvironmentSettings environmentSettings = CreateEnvironmentSettings(args);
+
+            CancellationToken cancellationToken = context.GetCancellationToken();
+
+            using AsyncMutex? entryMutex = await EnsureEntryMutex(args, environmentSettings, cancellationToken).ConfigureAwait(false);
+
+            try
+            {
+                //TODO:
+                //if (commandInput.IsHelpFlagSpecified)
+                //{
+                //    _telemetryLogger.TrackEvent(commandInput.CommandName + TelemetryConstants.HelpEventSuffix);
+                //}
+                using (Timing.Over(environmentSettings.Host.Logger, "Execute"))
+                {
+                    await HandleGlobalOptionsAsync(args, environmentSettings, cancellationToken).ConfigureAwait(false);
+                    return (int)await ExecuteAsync(args, environmentSettings, context).ConfigureAwait(false);
+                }
+            }
+            catch (Exception ex)
+            {
+                AggregateException? ax = ex as AggregateException;
+
+                while (ax != null && ax.InnerExceptions.Count == 1 && ax.InnerException is not null)
+                {
+                    ex = ax.InnerException;
+                    ax = ex as AggregateException;
+                }
+
+                Reporter.Error.WriteLine(ex.Message.Bold().Red());
+
+                while (ex.InnerException != null)
+                {
+                    ex = ex.InnerException;
+                    ax = ex as AggregateException;
+
+                    while (ax != null && ax.InnerExceptions.Count == 1 && ax.InnerException is not null)
+                    {
+                        ex = ax.InnerException;
+                        ax = ex as AggregateException;
+                    }
+
+                    Reporter.Error.WriteLine(ex.Message.Bold().Red());
+                }
+
+                Reporter.Error.WriteLine(ex.StackTrace.Bold().Red());
+                return 1;
+            }
+        }
+
+        public override IEnumerable<string> GetSuggestions(ParseResult? parseResult = null, string? textToMatch = null)
+        {
+            if (parseResult == null)
+            {
+                return base.GetSuggestions(parseResult, textToMatch);
+            }
+            TArgs args = ParseContext(parseResult);
+            IEngineEnvironmentSettings environmentSettings = CreateEnvironmentSettings(args);
+            return GetSuggestions(args, environmentSettings, textToMatch);
+        }
+
+        protected virtual IEnumerable<string> GetSuggestions(TArgs args, IEngineEnvironmentSettings environmentSettings, string? textToMatch)
+        {
+            return base.GetSuggestions(args.ParseResult, textToMatch);
+        }
+
+        protected IEngineEnvironmentSettings CreateEnvironmentSettings(TArgs args)
+        {
+            string? outputPath = (args as InstantiateCommandArgs)?.OutputPath;
```

Comment:
can we do any better? that's already broken

- reparse it inline to get output path?
- add output path as global option?
- recreate env settings before instantiating?

Reply:
Feels more like there is problem with `GlobalJsonFileExistsInPath` or `TryGetHostParamDefault`...
1st `GlobalJsonFileExistsInPath` should probably be Macro that takes Output folder as input.
2nd Maybe `TryGetHostParamDefault` needs access to existing Symbols/Output